tests/unit/windows/TestUwbAppConfiguration.cxx

```diff
@@ -114,8 +135,316 @@ TEST_CASE("UwbAppConfiguration performs allocation for contained value correctly
+        const auto& ddiParams = testStruct.DdiParameters();
+        const uint8_t* appConfigParamsBytes = reinterpret_cast<const uint8_t*>(&ddiParams.appConfigParams[0]);
+
+        REQUIRE(ddiParams.sessionId == expectedSessionId);
+        REQUIRE(ddiParams.appConfigParamsCount == std::size(parameters));
+        REQUIRE(ddiParams.size == parameters[0]->DdiSize() + parameters[1]->DdiSize() + offsetof(UWB_SET_APP_CONFIG_PARAMS, appConfigParams[0]));
+        REQUIRE(std::memcmp(appConfigParamsBytes, parameters[0]->DdiBuffer().data(), parameters[0]->DdiSize()) == 0);
+        REQUIRE(std::memcmp(&appConfigParamsBytes[parameters[0]->DdiSize()], parameters[1]->DdiBuffer().data(), parameters[1]->DdiSize()) == 0);
+    }
+}
+
+TEST_CASE("GenerateUwbSetAppConfigParameterDdi works", "[basic]")
+{
+    using windows::devices::uwb::GenerateUwbSetAppConfigParameterDdi;
+    using namespace ::uwb::protocol::fira;
+
+    SECTION("enum works")
+    {
+        using namespace windows::devices;
+
+        constexpr DeviceRole roleExpected{ DeviceRole::Initiator };
+
+        UwbSessionData sessionData;
+        sessionData.sessionId = 1;
+        sessionData.uwbConfiguration = UwbConfiguration::Create().DeviceRole(roleExpected);
+        sessionData.uwbConfigurationAvailable = true;
+
+        auto params = GenerateUwbSetAppConfigParameterDdi(sessionData);
+        const auto& ddiParams = params.DdiParameters();
+
+        REQUIRE(ddiParams.sessionId == sessionData.sessionId);
+        REQUIRE(ddiParams.appConfigParamsCount == 1);
+
+        const auto param = ddiParams.appConfigParams[0];
+        const auto sessionUwbMap = sessionData.uwbConfiguration.GetValueMap();
+
+        std::vector<std::unique_ptr<IUwbAppConfigurationParameter>> parameters;
+
+        for (const auto& [parameterTag, parameterValueVariant] : sessionUwbMap) {
+            const auto ddiType = windows::devices::detail::AppConfigUwbConfigurationTagMap.at(parameterTag);
+            REQUIRE(ddiType == param.paramType);
+
+            // the following doesn't want to compile
+            // parameters.push_back(std::make_unique<UwbAppConfigurationParameter<std::underlying_type_t<ParameterValueT>>>(notstd::to_underlying(parameterValueVariant), parameterTag));
+
+            parameters.push_back(std::make_unique<UwbAppConfigurationParameter<DeviceRole>>(roleExpected, parameterTag));
+
+            UwbSetAppConfigurationParameters testStruct{ parameters, sessionData.sessionId };
+
+            const auto& ddiParams = testStruct.DdiParameters();
+
+            REQUIRE(ddiParams.sessionId == sessionData.sessionId);
+            REQUIRE(ddiParams.appConfigParamsCount == std::size(parameters));
+            REQUIRE(ddiParams.size == parameters[0]->DdiSize() + offsetof(UWB_SET_APP_CONFIG_PARAMS, appConfigParams[0]));
+            REQUIRE(std::memcmp(ddiParams.appConfigParams, parameters[0]->DdiBuffer().data(), parameters[0]->DdiSize()) == 0);
+        }
+    }
+
+    SECTION("mac address works")
```

Comment:
This is good for a short mac.

We also need to test an extended mac. Can use the `UwbConfiguration::Builder::SetMacAddressController()` function to do that, and the parameter tag is `UwbConfiguration::ParameterTag::ControllerMacAddress`.